model/waterfall.go

```diff
@@ -0,0 +1,168 @@
+package model
+
+import (
+	"context"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/build"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+const (
+	DefaultWaterfallVersionLimit = 5
+	MaxWaterfallVersionLimit     = 300
```

Comment:
Is this going to be used in a later PR? I don't see it used right now.

300 also seems like a lot, do we expect we'll have to return 300 versions + all their BVs/tasks? Or is it because we anticipate the filters might need to search through a large number of versions?

Reply:
I used this value as suggested to cap user input above! But you are correct, the primary use case is to cap a search when supplied with a filter that has no results. We already use this pattern like so: https://spruce.mongodb.com/commits/evergreen-ui?buildVariants=foo&view=FAILED